scheduler.go

```diff
@@ -932,21 +932,39 @@ func (s *Scheduler) doCommon(jobFun interface{}, params ...interface{}) (*Job, e
 		return nil, job.error
 	}
 
-	typ := reflect.TypeOf(jobFun)
-	if typ.Kind() != reflect.Func {
+	val := reflect.ValueOf(jobFun)
+	for val.Kind() == reflect.Ptr {
+		val = val.Elem()
+	}
+
+	if val.Kind() != reflect.Func {
 		// delete the job for the same reason as above
 		s.RemoveByReference(job)
 		return nil, ErrNotAFunction
 	}
 
-	fname := getFunctionName(jobFun)
+	var fname string
+	if val == reflect.ValueOf(jobFun) {
+		fname = getFunctionName(jobFun)
+	} else {
+		fname = getFunctionNameOfPointer(jobFun)
+	}
+
 	if job.funcName != fname {
 		job.function = jobFun
+		if val != reflect.ValueOf(jobFun) {
+			job.function = reflect.ValueOf(jobFun).Elem().Interface()
```

Comment:
val would already be the reflect valueOf and the Elem if it's a pointer
```suggestion
			job.function = val.Interface()
```